plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestSingletonIdentityCacheMapping.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.plugin.jdbc.IdentityCacheMapping.IdentityCacheKey;
+import io.trino.spi.security.BasicPrincipal;
+import io.trino.spi.security.ConnectorIdentity;
+import io.trino.spi.security.SelectedRole;
+import io.trino.testing.TestingConnectorSession;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.trino.spi.security.SelectedRole.Type.ROLE;
+import static io.trino.spi.session.PropertyMetadata.stringProperty;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSingletonIdentityCacheMapping
```

Comment:
i do strongly feel it's an overkill.
it doesn't really prevent regressions (assuming no-one ever intentionally breaks equals() impl), or similar problems in the future